plugins/fr.inria.diverse.melange.ui/src/main/java/fr/inria/diverse/melange/ui/scope/MelangeScopeProvider.xtend

```diff
@@ -0,0 +1,116 @@
+package fr.inria.diverse.melange.ui.scope
+
+import com.google.inject.Inject
+import fr.inria.diverse.melange.ast.ModelingElementExtensions
+import fr.inria.diverse.melange.metamodel.melange.ClassBinding
+import fr.inria.diverse.melange.metamodel.melange.Import
+import fr.inria.diverse.melange.metamodel.melange.Language
+import fr.inria.diverse.melange.metamodel.melange.MelangePackage
+import fr.inria.diverse.melange.metamodel.melange.Merge
+import fr.inria.diverse.melange.metamodel.melange.PackageBinding
+import fr.inria.diverse.melange.metamodel.melange.Slice
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.xtext.scoping.IScope
+import org.eclipse.xtext.scoping.Scopes
+import org.eclipse.xtext.scoping.impl.AbstractDeclarativeScopeProvider
+import org.eclipse.xtext.scoping.impl.SimpleScope
+import org.eclipse.xtext.ui.resource.XtextResourceSetProvider
+import org.eclipse.xtext.naming.IQualifiedNameProvider
+
+class MelangeScopeProvider extends AbstractDeclarativeScopeProvider{
+
+    @Inject extension ModelingElementExtensions
+    @Inject XtextResourceSetProvider rsProvider
+    @Inject IQualifiedNameProvider prov
+
+    def dispatch IScope getScope(EObject context, EReference reference) {
+        return super.getScope(context, reference)
+    }
+
+    def dispatch IScope getScope(Language language, EReference ref){
+        if (ref.featureID == MelangePackage.LANGUAGE__OPERATORS) {
+        	val owning = language.name
+            val basic = super.getScope((language as EObject), ref).allElements
```

Comment:
Needed to force the dispatcher to call the getScope method for EObject. We don't want it to re-call the getScope for melange over and over again.